src/main/java/io/ifar/archive/ArchiveApplication.java

```diff
@@ -69,7 +69,8 @@ public void run(ArchiveConfiguration configuration, Environment environment) thr
 
         ArchiveListener listener = new ArchiveListener(s3Client,
                 configuration.getSeedBrokers(), workUnitPath, kafkaMessagePartitioner, configuration.getMaxNumParallelWorkers(),
-                environment.metrics(), configuration.getTopicConfiguration(), configuration.getDefaultBucket());
+                environment.metrics(), configuration.getTopicConfiguration(), configuration.getErrorLimitConfiguration(),
+                configuration.getDefaultBucket());
```

Comment:
I know @prb has different opinion, but to me passing full configuration object might make more sense at this point; called method can then access information it wants. Given that there's like 4 or more pieces of information it needs, and since Config object itself is a POJO and does not make testing any more complicated.